File: src/com/libill/demos/activity/SimpleCursorAdapterActivity.java
package com.libill.demos.activity;

import android.app.Activity;
import android.database.Cursor;
import android.os.Bundle;
import android.provider.Contacts.People;
import android.support.v4.widget.SimpleCursorAdapter;
import android.widget.ListAdapter;
import android.widget.ListView;

public class SimpleCursorAdapterActivity extends Activity {
	ListView listView;
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //获得一个指向系统通讯录数据库的Cursor对象获得数据来源
        Cursor cur = getContentResolver().query(People.CONTENT_URI, null, null, null, null);
        startManagingCursor(cur);
        //实例化列表适配器
        listView = new ListView(this);
        ListAdapter adapter = new SimpleCursorAdapter(this, android.R.layout.simple_list_item_1, cur, new String[] {People.NAME}, new int[] {android.R.id.text1});
        listView.setAdapter(adapter);
        setContentView(listView);
    }
}

File: app/src/main/java/com/libill/demos/activity/SimpleCursorAdapterActivity.kt
package com.libill.demos.activity

import android.R
import android.os.Bundle
import android.provider.Contacts
import android.widget.ListAdapter
import android.widget.ListView
import android.widget.Toast
import androidx.cursoradapter.widget.SimpleCursorAdapter
import androidx.lifecycle.lifecycleScope
import com.libill.demos.base.BaseActivity
import com.libill.demos.util.requestReadContactsPermission
import kotlinx.coroutines.launch

class SimpleCursorAdapterActivity : BaseActivity() {
    var listView: ListView? = null

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        listView = ListView(this)
        setContentView(listView)
    }

    override fun onResume() {
        super.onResume()
        lifecycleScope.launch {
            val granted = requestReadContactsPermission("需要读取通讯录权限")
            if (granted) {
                setData()
            } else {
                Toast.makeText(this@SimpleCursorAdapterActivity, "权限被拒绝", Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun setData() {
        //获得一个指向系统通讯录数据库的Cursor对象获得数据来源
        val cur = contentResolver.query(Contacts.People.CONTENT_URI, null, null, null, null)
        startManagingCursor(cur)
        //实例化列表适配器
        val adapter: ListAdapter = SimpleCursorAdapter(
            this, R.layout.simple_list_item_1, cur, arrayOf(Contacts.People.NAME), intArrayOf(
                R.id.text1
            )
        )
        listView!!.adapter = adapter
    }
}
